### PR TITLE
feat(ledger): Add /volumes endpoint with PIT and OOT filter

### PR DESCRIPTION
**Pull request by agourdel**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new API endpoint for retrieving volumes with their corresponding balances, enhancing the ledger's functionality.
	- Added pagination and filtering support for volume retrieval, allowing for more efficient data access and manipulation.
- **Tests**
	- Implemented a comprehensive test suite for the new volume retrieval functionality, ensuring reliability and robust performance.
- **Refactor**
	- Improved filtering functionality by splitting and refining existing filter methods, facilitating more precise and versatile data queries.
- **Chores**
	- Updated SDK versions and configurations to support the latest changes and improvements.

### CODE REVIEW

File: components/ledger/internal/api/v2/utils.go [hunk 1]
```diff
@@ -42,6 +75,20 @@ func getPITFilterWithVolumes(r *http.Request) (*ledgerstore.PITFilterWithVolumes
 	}, nil
 }
 
+func getPITOOTFilterForVolumes(r *http.Request) (*ledgerstore.PITFilterForVolumes, error) {
+	pit, err := getPITOOTFilter(r)
+	if err != nil {
+		return nil, err
+	}
+
+	UseInsertionDate := sharedapi.QueryParamBool(r, "use_insertion_date")
```
[the remark this answers is not in the document]
Comment (gfyrag) [reply components/ledger/internal/api/v2/utils.go:84]: (for the var)

[the remark this answers is not in the document]
Comment (gfyrag) [reply components/ledger/internal/api/v2/utils.go:84]: and the query param sorry

File: libs/go-libs/bun/bunpaginate/pagination_offset.go [hunk 2]
```diff
@@ -56,3 +56,53 @@ func UsingOffset[Q any, T any](ctx context.Context, sb *bun.SelectQuery, query O
 		Data:     ret,
 	}, nil
 }
+
+func UsingOffsetWithoutModel[Q any, T any](ctx context.Context, sb *bun.SelectQuery, query OffsetPaginatedQuery[Q], builders ...func(query *bun.SelectQuery) *bun.SelectQuery) (*Cursor[T], error) {
```
Comment (gfyrag) [inline libs/go-libs/bun/bunpaginate/pagination_offset.go:60]: Maybe factorize this with the original function?

**Review by gfyrag — APPROVED** [review]

File: components/ledger/internal/storage/ledgerstore/utils.go [hunk 3]
```diff
@@ -54,6 +54,21 @@ func paginateWithOffset[FILTERS any, RETURN any](s *Store, ctx context.Context,
...
+	q *bunpaginate.OffsetPaginatedQuery[FILTERS], builders ...func(query *bun.SelectQuery) *bun.SelectQuery) (*bunpaginate.Cursor[RETURN], error) {
+
+	//var ret RETURN
+	query := s.bucket.db.NewSelect()
+	for _, builder := range builders {
+		query = query.Apply(builder)
+	}
+	//if query.GetModel() == nil && query.GetTableName() == "" {
+	//	query = query.Model(ret)
+	//}
```
Comment (Dav-14) [inline components/ledger/internal/storage/ledgerstore/utils.go:67]: Is this needed

File: components/ledger/openapi.yaml [hunk 4]
```diff
@@ -2161,6 +2161,58 @@ paths:
...
+          schema:
+            type: string
+            format: date-time
+        - name: oot
+          in: query
+          required: false
+          schema:
+            type: string
+            format: date-time
+        - name: use_insertion_date
```
Comment (gfyrag) [inline components/ledger/openapi.yaml:2191]: Can you rename to `useInsertionDate` plz?

Comment (agourdel) [reply components/ledger/openapi.yaml:2191]: Alors, pour 'use_insertion_date' j'avoue avoir pris tel quel la variable pour la requête /aggregated/balances il me semble.

Tu veux que je la change aussi ?

**Pull request closed without merging**